domain-model-assistant/Assets/Components/Scripts/Diagram.cs

```diff
@@ -11,419 +11,549 @@
+    }
 
-  /// <summary>
-  /// Adds a node to the diagram.
-  /// </summary>
-  public bool AddNode(GameObject node)
-  {
-    if (compartmentedRectangles.Contains(node))
+    /// <summary>
+    /// Returns the list of compartmented rectangles.
+    /// </summary>
+    public List<GameObject> GetCompartmentedRectangles()
     {
-      return false;
+        return compartmentedRectangles;
     }
-    compartmentedRectangles.Add(node);
-    node.GetComponent<CompartmentedRectangle>().SetDiagram(this.gameObject);
-    return true;
-  }
-
-  public bool RemoveNode(GameObject node)
-  {
-    if (compartmentedRectangles.Contains(node))
+
+    /// <summary>
+    /// Activates the default mode.
+    /// </summary>
+    public void ActivateDefaultMode()
     {
-      compartmentedRectangles.Remove(node);
-      return true;
+        if (_isWebGl)
+        {
+            ResetCursor();
+        }
+        Debug.Log("Activating default mode");
+        _currentMode = CanvasMode.Default;
     }
-    return false;
-  }
-
-  /// <summary>
-  /// Returns the list of compartmented rectangles.
-  /// </summary>
-  public List<GameObject> GetCompartmentedRectangles()
-  {
-    return compartmentedRectangles;
-  }
-
-  /// <summary>
-  /// Activates the default mode.
-  /// </summary>
-  public void ActivateDefaultMode()
-  {
-    if (_isWebGl)
+
+    public void EnterAddClassMode()
     {
-      ResetCursor();
+        if (_isWebGl)
+        {
+            SetCursorToAddMode();
+        }
+        Debug.Log("Entering Add class mode");
+        _currentMode = CanvasMode.AddingClass;
     }
-    Debug.Log("Activating default mode");
-    _currentMode = CanvasMode.Default;
-  }
 
-  public void EnterAddClassMode()
-  {
-    if (_isWebGl)
+    /// <summary>
+    /// Called by AddClassAction when the AddClass button is pressed.
+    /// </summary>
+    public void AddClassButtonPressed()
     {
-      SetCursorToAddMode();
+        if (_currentMode == CanvasMode.AddingClass)
+        {
+            ActivateDefaultMode();
+        }
+        else
+        {
+            EnterAddClassMode();
+        }
     }
-    Debug.Log("Entering Add class mode");
-    _currentMode = CanvasMode.AddingClass;
-  }
-
-  /// <summary>
-  /// Called by AddClassAction when the AddClass button is pressed.
-  /// </summary>
-  public void AddClassButtonPressed()
-  {
-    if (_currentMode == CanvasMode.AddingClass)
+
+    /// <summary>
+    /// Perform the debug action specified in the body upon the Debug button getting clicked.
+    /// </summary>
+    public void DebugAction()
     {
-      ActivateDefaultMode();
+        Debug.Log("Debug button clicked!");
+        //LoadData();
+        GetCompartmentedRectangles()[0].GetComponent<CompartmentedRectangle>().GetHeader().GetComponent<InputField>()
+          .text = "Rabbit";
     }
-    else
+
+    //When pressing on canvas close popumenu and attributeclosebuttons
+    public void CloseMenus()
     {
-      EnterAddClassMode();
+
```

Comment:
Nothing is being done here so far. Add a `TODO` to make this more clear and to let others know what to expect here.

Reply:
Done